extensions/amp-story-panning-media/0.1/amp-story-panning-media.js

```diff
@@ -32,55 +35,105 @@ export class AmpStoryPanningMedia extends AMP.BaseElement {
     /** @private {?Element} */
     this.element_ = element;
 
-    /** @private {?Element} */
-    this.image_ = null;
+    /** @private {?Element} The element that is transitioned. */
+    this.ampImgEl_ = dev().assertElement(
+      this.element_.querySelector('amp-img')
+    );
 
-    /** @private {?string} */
-    this.x_ = null;
+    /** @public {?string} Sent to siblings to update their position. */
+    this.x = this.element_.getAttribute('x') || '0%';
 
-    /** @private {?string} */
-    this.y_ = null;
+    /** @public {?string} Sent to siblings to update their position. */
+    this.y = this.element_.getAttribute('y') || '0%';
 
-    /** @private {?string} */
-    this.zoom_ = null;
+    /** @public {?string} Sent to siblings to update their position. */
+    this.zoom = this.element_.getAttribute('zoom') || '1';
+
+    /** @private {Array<Element>} */
+    this.siblings_ = [];
   }
 
   /** @override */
   buildCallback() {
-    this.x_ = this.element_.getAttribute('x') || '0%';
-    this.y_ = this.element_.getAttribute('y') || '0%';
-    this.zoom_ = this.element_.getAttribute('zoom') || '1';
+    this.getSiblings();
+
+    // Initialize all services before proceeding
+    return Promise.all([
+      Services.storyStoreServiceForOrNull(this.win).then((storeService) => {
+        storeService.subscribe(StateProperty.CURRENT_PAGE_ID, (currPageId) => {
```

Comment:
Should add the optional 3rd parameter `true` so it fetches the current page id when initialized

Reply:
This does appear to be called when landing on the story. 
when adding the `callToInitialize` this method gets called twice when landing on the story.

It definitely makes it more readable but does it have any other benefits if it's running anyway on page load?

Reply:
This callback will only be called on PAGE_ID change, you should definitely have the `callToInitialize` set to true if you want this to work without any race condition.
If it is called twice, it might be because you get the initial value (probably null or empty string) which you can ignore.